### ISSUE TITLE
Implement the new C23 standard

### ISSUE THREAD

**Issue by chuggafan** [LADSoft/OrangeC#685]
https://open-std.org/JTC1/SC22/WG14/www/docs/n3047.pdf

We actually have a lot of code that works for this as this is mostly an update to get C++ features in C, however the main workload I see being a problem will be `#embed`, which is a preprocessor addition. I am using a draft that shows all the features added as well.

Edit: not sure what milestone this belongs in, just not in the current one, maybe 4th milestone would be a good idea?

**Comment by LADSoft** [comment]
yeah if it is really mostly C++ stuff ported to C it will be fairly easy, although, we might want to consolidate the C compiler version checking into a single variable...   #embed doesn't look hard, I write programs that generate code like that all the time.   We can resurrect the AES example I had and move the encryption/decryption tables into binary files as a test program lol...

I'll go ahead and put it in milestone 4 :smile:

**Comment by LADSoft** [comment]
I'll be working on this for a while.    The big thing left is _BitInt but there are a lot of smaller issues as well...

**Comment by LADSoft** [comment]
I'm through the BITINT stuff and worked on a bunch of the more piddly stuff.  I have two or three issues left to do then we will be mostly done other than the #953
